src/MyConnection.h

```diff
@@ -6,6 +6,10 @@
 #include <boost/noncopyable.hpp>
 #include <boost/shared_ptr.hpp>
 
+const char* as_api_string(const Rcpp::Nullable<std::string>& cpp_string) {
```

Comment:
Thanks. We tried that, see the discussion https://github.com/rstats-db/RMySQL/pull/90#discussion_r50238535 and below. (I should have posted this link a bit earlier, but just found it now.)